src/read.c

```diff
@@ -3607,198 +3559,276 @@ static avifProperty * avifDecoderItemAddProperty(avifDecoderItem * item, const a
+
+    // Make sure all metadata and coded chunks fit into the 'meta' box whose size is rawLen.
     // There should be no missing nor unused byte.
-    AVIF_CHECKERR(avifROStreamRemainingBytes(&s) ==
-                      (uint64_t)extendedMetaSize + iccDataSize + alphaItemDataSize + colorItemDataSize + exifSize + xmpSize,
-                  AVIF_RESULT_BMFF_PARSE_FAILED);
 
-    // Store and update the offset for the following item extents and properties.
-    // The extendedMeta field is parsed after creating the items defined by the MinimizedImageBox
-    // so the stream s cannot be used for keeping track of the position.
-    uint64_t offset = rawOffset + avifROStreamOffset(&s) + extendedMetaSize;
+    AVIF_CHECKERR(avifROStreamRemainingBytes(s) == (uint64_t)iccDataSize + alphaItemDataSize + mainItemDataSize + exifDataSize + xmpDataSize,
+                  AVIF_RESULT_BMFF_PARSE_FAILED);
 
-    // Create the items and properties generated by the MinimizedImageBox.
-    // The MinimizedImageBox always creates 8 properties for specification easiness.
+    // Create the items and properties generated by the MetaBox with version 1.
+    // The MetaBox with version 1 always creates 8 properties for specification easiness.
     // Use FreeSpaceBoxes as no-op placeholder properties when necessary.
     // There is no need to use placeholder items because item IDs do not have to
     // be contiguous, whereas property indices shall be 1, 2, 3, 4, 5 etc.
 
     meta->primaryItemID = 1;
     avifDecoderItem * colorItem;
     AVIF_CHECKRES(avifMetaFindOrCreateItem(meta, meta->primaryItemID, &colorItem));
-    memcpy(colorItem->type, "av01", 4);
+    memcpy(colorItem->type, infeType, 4);
     colorItem->width = width;
     colorItem->height = height;
-    colorItem->miniPixelFormat = yuvFormat;
-    colorItem->miniChromaSamplePosition = yuvChromaSamplePosition;
+    colorItem->metaV1PixelFormat = chromaSubsampling == 0   ? AVIF_PIXEL_FORMAT_YUV400
+                                   : chromaSubsampling == 1 ? AVIF_PIXEL_FORMAT_YUV420
+                                   : chromaSubsampling == 2 ? AVIF_PIXEL_FORMAT_YUV422
+                                                            : AVIF_PIXEL_FORMAT_YUV444;
+    if (colorItem->metaV1PixelFormat == AVIF_PIXEL_FORMAT_YUV422) {
+        // In AV1, the chroma_sample_position syntax element is not present for the YUV 4:2:2 format.
+        // Assume that AV1 uses the same 4:2:2 chroma sample location as HEVC and VVC (colocated).
+        AVIF_CHECKERR(!isHorizontallyCentered, AVIF_RESULT_BMFF_PARSE_FAILED);
+        // isVerticallyCentered: Ignored unless chroma_subsampling is 1.
+        colorItem->metaV1ChromaSamplePosition = AVIF_CHROMA_SAMPLE_POSITION_UNKNOWN;
+    } else if (colorItem->metaV1PixelFormat == AVIF_PIXEL_FORMAT_YUV420 && isHorizontallyCentered) {
+        // There is no way to describe this with AV1's chroma_sample_position enum besides CSP_UNKNOWN or CSP_RESERVED.
```

Comment:
Nit/optional: The "or CSP_RESERVED" part is a little confusing. I suggest removing it. Perhaps we could add a second sentence "There is a proposal to assign the reserved value 3 (CSP_RESERVED) to the center chroma sample position."